File: WOT-iOS/WOT-iOS/core/requests/WOTWebRequest/Vehicles/WOTWEBRequestTankVehicles.m
//
//  WOTWEBRequestTankVehicles.m
//  WOT-iOS
//
//  Created by Pavel Yeshchyk on 6/19/15.
//  Copyright (c) 2015 Pavel Yeshchyk. All rights reserved.
//

#import "WOTWEBRequestTankVehicles.h"

@implementation WOTWEBRequestTankVehicles

- (NSDictionary *)query {
    
    NSMutableDictionary *result = [[NSMutableDictionary alloc] init];
    result[WOT_KEY_APPLICATION_ID] = [NSString valueOrSpaceString:self.hostConfiguration.applicationID];
    result[WOT_KEY_FIELDS] = [NSString valueOrSpaceString:self.args[WOT_KEY_FIELDS]];
    result[WOT_KEY_TANK_ID] = [NSString valueOrSpaceString:self.args[WOT_KEY_TANK_ID]];
    return result;
}

- (NSString *)path {
    
    return @"wot/encyclopedia/vehicles/";
}


- (NSString *)method {
    
    return @"POST";
}

@end

File: WOT-iOS/WOTData/WOTData/WebServices/requests/WOTWEBRequestTankVehicles.swift
//
//  WOTWEBRequestTankVehicles.swift
//  WOTData
//
//  Created by Pavel Yeshchyk on 1/14/20.
//  Copyright © 2020 Pavel Yeshchyk. All rights reserved.
//

import Foundation

@objc
public class WOTWEBRequestTankVehicles: WOTWEBRequest, WOTModelServiceProtocol {

    override public var method: String { return "POST" }

    @available(*, deprecated, message: "TO be refactored")
    @objc
    public class func modelClassName() -> String {
        return NSStringFromClass(Vehicles.self)
    }

    @available(*, deprecated, message: "TO be refactored")
    @objc
    public func instanceModelClass() -> AnyClass? {
        return NSClassFromString( type(of: self).modelClassName() )
    }

//    override public var query: [AnyHashable : Any] {
////        let fields = self.args?.escapedValue(forKey: WGWebQueryArgs.fields) ?? ""
//        let fields = ""
//        return [WGWebQueryArgs.application_id: self.hostConfiguration.applicationID,
//                WGWebQueryArgs.fields: fields]
//    }
    
    override public var path: String {
        return "/wot/encyclopedia/vehicles/"
    }
}
